### PR TITLE
Harsh/appointment data change

### LINKED ISSUE #97: Feature - [User] Book appointment changes
**Is your feature request related to a problem? Please describe.**
- Show the time ascending order.
- Book appointment in db as per new changes

### Check List

- [x] I have linked my PR with this task
- [x] I have added comment after completing this task
- [x] I have followed all [guidelines](https://github.com/jaykishan-sewak/doctor-appoint-app/tree/main) for code

### LINKED ISSUE #99: Bug - [doctor] Profile screen crash
**Describe the bug**
In Update profile screen click on holiday button select date click on ok app is crash.

**To Reproduce**
1. Login with doc1.
2. Go to profile
3. Click on edit icon
4. Click on holiday button 
5. Select date click on ok app is crash.

**Expected behavior**
App not crash any condition.

### Check List

- [ ] I have linked my PR with this task
- [ ] I have added comment after completing this task
- [ ] I have followed all [guidelines](https://github.com/jaykishan-sewak/doctor-appoint-app/tree/main) for code

### LINKED ISSUE #100: Bug - [Doctor] Update profile screen
**Describe the bug**
- Not able to add time every time show **already time is selected toast**

**To Reproduce**
1. Login with doc1.
2. Go to profile
3. Click on edit move to profile screen
4. Click on add time 
5. Click on select time and select any time show **already time is selected toast**

**Expected behavior**
Validation should be work properly.

### Check List

- [ ] I have linked my PR with this task
- [ ] I have added comment after completing this task
- [ ] I have followed all [guidelines](https://github.com/jaykishan-sewak/doctor-appoint-app/tree/main) for code

### LINKED ISSUE #114: Feature - [Doctor] Doctor update profile screen
**Is your feature request related to a problem? Please describe.**
- show all details when doctor move to update profile screen.
- if doctor already added holiday,time and weekoff  then that data should be show in ui.

- Not able to add weekofflist.

- Go to profile tab. Click on edit update some field data then update successfully but move to Appointments tab. Please stay in profile tab.

- User view any doctor then app is crash.

### Check List

- [x] I have linked my PR with this task
- [x] I have added comment after completing this task
- [x] I have followed all [guidelines](https://github.com/jaykishan-sewak/doctor-appoint-app/tree/main) for code

### PR DESCRIPTION
**Pull request by HarshPanchalInfusion**
Update screen Show holiday, week off, available Time & update,delete

### CODE REVIEW

File: app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt [hunk 1]
```diff
@@ -845,9 +863,21 @@ class AddDoctorViewModel @Inject constructor(
...
+    private fun getDBWeekDayList(weekOffList: java.util.ArrayList<String>?) {
+        if (weekOffList?.isNotEmpty() == true) {
+            weekDayList.forEachIndexed { index, weekOffModel ->
+                weekOffList.forEachIndexed { i, s ->
+                    if (weekOffModel.dayName == s)
+                        weekDayList[index].isWeekOff = true
 
+                }
+            }
+        } else {
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt]: Please remove else part if no need

File: app/src/main/java/com/android/doctorapp/ui/doctor/UpdateDoctorProfileFragment.kt [hunk 2]
```diff
@@ -480,7 +516,7 @@ class UpdateDoctorProfileFragment :
     }
 
     private fun updateWeekOffRecyclerview(weekOffDayList: ArrayList<WeekOffModel>) {
-        weekOffDayAdapter = WeekOffDayAdapter(weekOffDayList,
+        /*weekOffDayAdapter = WeekOffDayAdapter(weekOffDayList,
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/ui/doctor/UpdateDoctorProfileFragment.kt]: Remove commented code if no need

File: app/src/main/java/com/android/doctorapp/ui/doctor/UpdateDoctorProfileFragment.kt [hunk 3]
```diff
@@ -493,15 +529,30 @@ class UpdateDoctorProfileFragment :
...
+            })*/
+
+
+        weekOffDayAdapter = WeekOffDayAdapter(weekOffDayList,
+            object : WeekOffDayAdapter.OnItemClickListener {
+                override fun onItemClick(item: WeekOffModel, position: Int) {
+                    tempWeekOffList.forEachIndexed { index, weekOffModel ->
+                        if (tempWeekOffList[index].dayName == item.dayName) {
+                            tempWeekOffList[index].isWeekOff = !tempWeekOffList[index].isWeekOff
+                        } else {
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/ui/doctor/UpdateDoctorProfileFragment.kt]: else part remove if no need

File: app/src/main/java/com/android/doctorapp/util/bindingAdapter/BindingAdapters.kt [hunk 4]
```diff
@@ -258,8 +258,9 @@ fun setString(textView: AppCompatTextView, specialities: List<String>?) {
 }
 
 @BindingAdapter("app:dateMonthStyle")
-fun setDateMonthStyle(textView: AppCompatTextView, originalTimeStr: String) {
-    textView.text = convertFullDateToDate(originalTimeStr)
+fun setDateMonthStyle(textView: AppCompatTextView, originalTimeStr: Date) {
+//    textView.text = convertFullDateToDate(originalTimeStr)
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/util/bindingAdapter/BindingAdapters.kt]: Remove unused code

File: app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt (hunk 1 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt]: Done

File: app/src/main/java/com/android/doctorapp/ui/doctor/UpdateDoctorProfileFragment.kt (hunk 2 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/ui/doctor/UpdateDoctorProfileFragment.kt]: Changes Done

File: app/src/main/java/com/android/doctorapp/ui/doctor/UpdateDoctorProfileFragment.kt (hunk 3 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/ui/doctor/UpdateDoctorProfileFragment.kt]: Changes Done

File: app/src/main/java/com/android/doctorapp/util/bindingAdapter/BindingAdapters.kt (hunk 4 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/util/bindingAdapter/BindingAdapters.kt]: Changes Done

**Review by vikas-infusion — APPROVED** [review]

**Pull request merged**